src/services/AcceptStakeRequestService.ts

```diff
@@ -0,0 +1,172 @@
+// Copyright 2019 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// ----------------------------------------------------------------------------
+
+import BigNumber from 'bignumber.js';
+import Web3 from 'web3';
+import Database from '../models/Database';
+import { StakeRequest, StakeRequestRepository } from '../models/StakeRequestRepository';
+import Observer from '../observer/Observer';
+import {
+  MessageType, MessageStatus, MessageDirection, MessageRepository, MessageAttributes,
+} from '../models/MessageRepository';
+
+import crypto = require('crypto');
+const hash = crypto.createHash('sha256');
+
+/**
+ * Class collects all non accepted stake requests on a trigger and accepts
+ * those stake requests in parallel.
+ */
+export default class AcceptStakeRequestService extends Observer {
+  /* Storage */
+
+  private web3: Web3;
+
+  private stakeRequestRepository: StakeRequestRepository;
+
+  private messageRepository: MessageRepository;
+
+
+  /* Public Functions */
+
+  public constructor(db: Database, web3: Web3) {
+    super();
+
+    this.web3 = web3;
+    this.stakeRequestRepository = db.stakeRequestRepository;
+    this.messageRepository = db.messageRepository;
+  }
+
+  public async notify(): Promise<void> {
+    const nonAcceptedStakeRequests = await this.collectNonAcceptedStakeRequests();
+    await this.acceptStakeRequests(nonAcceptedStakeRequests);
```

Comment:
If the notify is called multiple times, it will call `acceptStakeRequests` multiple times, 
We should have a locking mechanism (may be in DB table). What do you think?

Reply:
If we `await` call to finish, will it be sufficient?

Reply:
`await` will not be sufficient. 
We should discuss this on call.